apps/prairielearn/src/lib/grading.ts

```diff
@@ -53,6 +53,7 @@ const VariantForSubmissionSchema = VariantSchema.extend({
 type SubmissionDataForSaving = Pick<Submission, 'variant_id' | 'auth_user_id'> &
   Pick<Partial<Submission>, 'credit' | 'mode' | 'client_fingerprint_id'> & {
     submitted_answer: NonNullable<Submission['submitted_answer']>;
+    user_id: string | null;
```

Comment:
We are not storing `user_id` in the submissions table only a `authn_user_id`. Adding `user_id` to the type here which we will need to pass if we need to create an issue.

Reply:
I think this is OK. An alternative would be to add this as a separate parameter to functions that need it, but we're already bundling a bunch of context into this `submissionData`, so 🤷 

I think this can be made non-nullable.

Reply:
Made `user_id` non-nullable in [f33eebc](https://github.com/PrairieLearn/PrairieLearn/pull/10840/commits/f33eebcc2cd336dd68a9f4942e2d3abac4b6855b).